### PR TITLE
ENCD-4427-file-bucket-audit

### PR DESCRIPTION
**Pull request by keenangraham**
Adds audit to file that tells which bucket it should be in based on its status. Also adds update_bucket endpoint in order to update bucket name with external patch request.

### CODE REVIEW

File: src/encoded/audit/file.py [hunk 1]
```diff
@@ -349,7 +367,7 @@ def audit_duplicate_quality_metrics(value, system):
     'audit_paired_with': audit_paired_with,
     'audit_specifications': audit_file_format_specifications,
     'audit_controlled_by': audit_file_controlled_by,
-    'audit_duplicate_quality_metrics': audit_duplicate_quality_metrics,
+    'audit_duplicate_quality_metrics': audit_duplicate_quality_metrics
```
Comment (hitz) [inline src/encoded/audit/file.py]: I think PEP standard is to end with a ','?

Comment (keenangraham) [reply src/encoded/audit/file.py]: Fixed.

**Comment by keenangraham** [comment]
@gabdank Fixed.

File: buildout.cfg [hunk 2]
```diff
@@ -93,6 +93,8 @@ file_upload_profile_name = encoded-files-upload
 ontology_path = ${buildout:directory}/ontology.json
 external_aws_s3_transfer_allow = false
 external_aws_s3_transfer_buckets = ${buildout:directory}/.aws/direct-external-s3-list
+pds_private_bucket = encode-pds-private-dev
```
[the remark this answers is not in the document]
Comment (keenangraham) [reply buildout.cfg]: Yeah, we have to figure that out still.

File: src/encoded/audit/file.py [hunk 3]
```diff
@@ -377,6 +396,7 @@ def audit_file(value, system):
         for failure in function_dispatcher[function_name](value, system):
             yield failure
 
+
```
[the remark this answers is not in the document]
Comment (keenangraham) [reply src/encoded/audit/file.py]: Fixed.

File: buildout.cfg (hunk 2 above)
[the remark this answers is not in the document]
Comment (caseylitton) [reply buildout.cfg]: should just be able to add it to the candidate.cfg

File: src/encoded/types/file.py [hunk 4]
```diff
@@ -473,6 +482,40 @@ def set_status(self, new_status, request, parent=True):
...
+            # File object doesn't exist, leave it alone.
+            return (return_flag, None, None)
+        current_bucket = external.get('bucket')
+        current_key = external.get('key')
+        base_uri = 's3://{}/{}'
+        current_path = base_uri.format(current_bucket, current_key)
+        file_status = properties.get('status')
+        # Released restricted files should be in private bucket.
+        if file_status in self.private_s3_statuses or not self._should_set_object_acl():
+            if current_bucket == private_bucket:
```
Comment (caseylitton) [inline src/encoded/types/file.py]: the return flag_is already true so we should be able to just
```
if not current_bucket == private_bucket:
     return_flag = False
```
Same for the public block.  Save a few lines of code

File: buildout.cfg (hunk 2 above)
[the remark this answers is not in the document]
Comment (caseylitton) [reply buildout.cfg]: Now that we have the public buckets setup we should add the vars to the candidate.cfg

**Pull request merged**